src/sage/algebras/weyl_algebra.py

```diff
@@ -1123,3 +1076,475 @@ def _act_(self, g, x):
+                res += m[1]._repr_()
+            return res if res != '' else '1'
+        return repr_from_monomials(self.list(), term)
+
+    def _mul_(self, other):
+        """
+        Multiply ``self`` by ``other``
+
+        TESTS::
+
+            sage: W.<x> = InfiniteDifferentialWeylAlgebra(QQ)
+            sage: dx = W.differentials()
+            sage: dx[1]*(x[1]*x[2] + x[3])
+            x[1]*x[2]*dx[1] + x[3]*dx[1] + x[2]
+            sage: dx[132]*x[132] - x[132]*dx[132] == 1
+            True
+        """
+        out = {}
+        zero = self.parent().base_ring().zero()
+        zero_m = self.parent()._diff_index.one()
+        # multiply each pair of monomials
+        for ml in self._monomial_coefficients:
+            cl = self._monomial_coefficients[ml]
+            for mr in other._monomial_coefficients:
+                cr = other._monomial_coefficients[mr]
+                # apply derivative terms of ml to mr and simplify
+                cur = [((mr[0], zero_m), cl*cr)]
+                ldd = ml[1].dict()
+
+                for i in ldd:
+                    p = ldd[i]
+                    # applying dx[i] p times
+                    for _ in range(p):
+                        next = []
+                        for m, c in cur:
+                            diff = dict(m[1].dict())
+                            diff[i] = diff.get(i, zero) + 1
+                            next.append(((m[0], self.parent()._diff_index(diff)), c))
+                            # power rule if m has x[i] term
+                            poly = m[0].dict()
+                            if i in poly:
+                                c *= poly[i]
+                                poly[i] -= 1
+                                next.append(((self.parent()._var_index(poly), m[1]), c))
+                        cur = next
+
+                for m, c in cur:
+                    m = (ml[0] * m[0], m[1] * mr[1])
+                    out[m] = out.get(m, zero) + c
+                    if out[m] == zero:
+                        del out[m]
+        return self.__class__(self.parent(), out)
+
+    def __iter__(self):
+        """
+        Return an iterator of ``self``.
+
+        EXAMPLES::
+
+            sage: W.<x> = InfiniteDifferentialWeylAlgebra(QQ)
+            sage: dx = W.differentials()
+            sage: p = x[5] + dx[1]*x[1]
+            sage: list(p)
+            [((x[1], dx[1]), 1), ((x[5], 1), 1), ((1, 1), 1)]
+        """
+        return iter(self.list())
+
+    def list(self):
+        """
+        Return ``self`` as a list.
+
+        This list consists of pairs `(m, c)` where `m` is a pair of :class:`IndexedFreeAbelianMonoid<sage.monoids.indexed_free_monoid.IndexedFreeAbelianMonoid>`
+        elements indexing a basis element of ``self``, and `c` is the corresponding (nonzero) coefficient.
+        The list is sorted using graded lex order on the differentials,
+        followed by graded lex order on the polynomial generators.
```

Comment:
```suggestion
        This list consists of pairs `(m, c)` where `m` is a pair of
        :class:`~sage.monoids.indexed_free_monoid.IndexedFreeAbelianMonoid`
        elements indexing a basis element of ``self``, and `c` is the corresponding
        (nonzero) coefficient.  list is sorted using graded lex order on the
        differentials, followed by graded lex order on the polynomial generators.
```
It's possible I am over the 80 char/line here. Adjust as necessary (but leave the `:class:` part on its own line even if over since it can't be made shorter).